src/Core/Framework/Adapter/Twig/TwigEnvironment.php

```diff
@@ -34,14 +34,53 @@ public function compile(Node $node): string
         }
 
         $source = $this->compiler->compile($node)->getSource();
+        if ($this->shouldAddMacroResult($source)) {
+            $source = $this->addMacroResult($source);
+        }
 
         $replaces = [
             'CoreExtension::getAttribute(' => 'SwTwigFunction::getAttribute(',
-            'CoreExtension::callMacro(' => 'SwTwigFunction::callMacro(',
             'twig_escape_filter(' => 'SwTwigFunction::escapeFilter(',
             'use Twig\Environment;' => "use Twig\Environment;\nuse Shopware\Core\Framework\Adapter\Twig\SwTwigFunction;",
         ];
 
         return str_replace(array_keys($replaces), array_values($replaces), $source);
     }
+
+    private function shouldAddMacroResult(string $source): bool
+    {
+        return
+            str_contains($source, 'getTemplateForMacro')
+            && str_contains($source, 'SwTwigFunction::$macroResult =')
+            && (!str_contains($source, 'CoreExtension::getAttribute') || str_contains($source, 'ComparisonExtension'))
+        ;
+    }
+
+    private function addMacroResult(string $source): string
+    {
+        [$lines, $lineNumber] = $this->extractLinesAndNumber($source);
+        if (isset($lineNumber)) {
+            $callMacroResult = 'yield SwTwigFunction::$macroResult;';
```

Comment:
just wondering:
Is this important that this happens after the `getTemplateForMacro` call? Or can we do that in the compiled template and add the yield again in the return node 🤔